tests/integration/test_manifest_validation.py

```diff
@@ -0,0 +1,486 @@
+                )
+            }
+            submit_response = requests.post(
+                submit_url, headers=HEADERS, params=submit_params, files=submit_files
+            )
+            assert (
+                submit_response.status_code == 200
+            ), f"File submission was unsuccessful. Got {submit_response.status_code}"
+
+        # AND the manifest validation endpoint and parameters are given
+        url = os.path.join(setup_api, "model/validate")
+        params = {
+            "schema_url": "https://raw.githubusercontent.com/Sage-Bionetworks/schematic/develop/tests/data/example.model.jsonld",
+            "data_type": "MockComponent",
+            "data_model_labels": "class_label",
+            "restrict_rules": False,
+            "asset_view": "syn63596704",
+            "project_scope": "syn63582791",
+        }
+
+        # AND a list of expected warnings is given
+        expected_warnings = [
+            [
+                None,
+                "Check Recommended",
+                "Column Check Recommended is recommended but empty.",
+                None,
+            ],
+            [
+                None,
+                "Check Match at Least",
+                "Cross Manifest Validation Warning: There are no target columns to validate this manifest against for attribute: Check Match at Least, and validation rule: matchAtLeastOne Patient.PatientID set. It is assumed this is the first manifest in a series to be submitted, so validation will pass, for now, and will run again when there are manifests uploaded to validate against.",
+                None,
+            ],
+        ]
+
+        # AND a file to be uploaded for validation is defined
+        input_file = "mock_manifests/MockComponent-cross-manifest-2.csv"
+        input_file_path = helpers.get_data_path(input_file)
+        with open(input_file_path, "rb") as file_obj:
+            files = {
+                "file_name": (
+                    os.path.basename(input_file),
+                    file_obj,
+                    "text/csv",
+                )
+            }
+
+            # AND we make a POST request to validate the file
+            response = requests.post(url, headers=HEADERS, params=params, files=files)
+
+            # THEN we expect a successful response
+            assert (
+                response.status_code == 200
+            ), f"Should be 200 status code. Got {response.status_code}"
+
+            # AND the response should contain the expected warnings
+            content = response.content.decode("utf-8")
+            data = json.loads(content)
+            warnings = data.get("warnings", [])
+
+            for idx, expected_idx in zip(warnings, expected_warnings):
+                assert idx == expected_idx
+
+        # TODO: Turn this into a class attribute in case we need the Synapse instance
+        # in other places.
+        import synapseclient
+
+        syn = synapseclient.login()
```

Comment:
Instead of creating this yourself this is available as a fixture from contest.py:

```
@pytest.fixture(scope="class")
def syn(syn_token) -> Synapse:
    syn = Synapse()
    syn.login(authToken=syn_token, silent=True)
    return syn
```